core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -85,6 +85,9 @@ private[spark] class Executor(
     env.blockManager.initialize(conf.getAppId)
   }
 
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
```

Comment:
is the hostname really enough?  What about with multiple executors on the same host?  The `Heartbeat` already has the `executorId`, maybe that is all we need?